src/globus_sdk/scopes/builder.py

```diff
@@ -0,0 +1,161 @@
+import typing as t
+
+from .scope_definition import Scope
+
+ScopeBuilderScopes = t.Union[
+    None, str, t.Tuple[str, str], t.List[t.Union[str, t.Tuple[str, str]]]
+]
+
+
+class ScopeBuilder:
```

Comment:
With the introduction of the class `Scope`, I think the name `ScopeBuilder` becomes a bit confusing.
`ScopeBuilder` builds a `string`, not a `scope`.

Probably beyond the fold for this change, but we should probably rethink what this class is responsible for & name it accordingly.
Ideas from the top of my head:
 * `ScopeRegistry`
 * `ScopeStringRegistry`
 * `ScopeStringFormatter`
 * `ScopeDict`

Reply:
I want to address this in SDKv4 by renaming the object and making its members `Scope` objects. That would remove any need for `make_mutable` as a method, and provide a more consistent view that what the SDK provides are high-level helper objects which _render_ as strings.

The fact that the `ScopeBuilder` objects are attached to and re-exposed through client classes makes this fiddly to even attempt prior to v4.

Reply:
Yeah, I agree with that plan.

Just a side-effect I thought of while reading & wanted to make sure that opinion was voiced.